chunks.go

```diff
@@ -216,6 +220,8 @@ func (w *chunkWriter) WriteChunks(chks ...*ChunkMeta) error {
 		chk.Chunk = nil
 	}
 
+	// XXX: Shouldn't the CRC32 length be included in maxLen?
+	// XXX: Shouldn't w.n be incremented here?
```

Comment:
Hurdur, of course (about the n increment).

But if we use `maxLen` to determine whether we still have space in the current segment, but we also write the CRC32 to the segment buffer, shouldn't it be included in that size calculation? Otherwise we'd need to preallocate to `segmentSize + 4`? But I'm probably just jet-lagged and dense again :)